XiEditor/Search.swift

```diff
@@ -0,0 +1,208 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+import Swift
+
+class FindViewController: NSViewController, NSTextFieldDelegate {
+
+    @IBOutlet weak var searchField: NSSearchField!
+    @IBOutlet weak var navigationButtons: NSSegmentedControl!
+    @IBOutlet weak var doneButton: NSButton!
+    @IBOutlet weak var ignoreCaseMenuItem: NSMenuItem!
+    @IBOutlet weak var wrapAroundMenuItem: NSMenuItem!
+
+    var optionIgnoreCase: Bool = true
+    var optionWrapAround: Bool = true
+
+    var editViewController: EditViewController!
```

Comment:
this should be a `weak var`, or we have a reference cycle. (Not the end of the world, in this case).